src/plugins/unified_search/public/search_bar/search_bar.tsx

```diff
@@ -528,12 +528,27 @@ class SearchBarUI<QT extends (Query | AggregateQuery) | Query = Query> extends C
     ) : undefined;
 
     let filterBar;
+    const esqlIndexPatternMap = new Map<string, boolean>();
+    const indexPatterns = this.props.indexPatterns
+      ? this.props.indexPatterns.reduce((collector, dataView) => {
+          if (dataView.type !== 'esql') {
+            collector.push(dataView);
+          } else {
+            const indexPattern = dataView.getIndexPattern();
+            if (!esqlIndexPatternMap.has(indexPattern)) {
+              esqlIndexPatternMap.set(indexPattern, true);
+              collector.push(dataView);
+            }
+          }
+          return collector;
+        }, [] as DataView[])
```

Comment:
Having a second look, and hands on is always recommended. yes, this should resolve #174736, found no caching issue 🎉

Reply:
Well, I found a (minor) caching issue, it seems, when there are multiple data view of same index pattern, there should be 1 requests uncached, the rest are cached, but it seems there can be more then one uncached requests ... however, all subsequent request are cached
![esql_test_-_Elastic](https://github.com/elastic/kibana/assets/463851/bcb920ba-3416-46f1-a61e-9f0e67de686c)

Reply:
@kertal This is a dashboard with 5 charts with the same esql data source?

Reply:
yes, but well, it's an initial data view loading, after a refreshing of cache https://mattkime-pr-172622-mark-esql-data-view-as-such.kbndev.co/app/dashboards#/view/09f7781d-78af-4b36-8eb0-f6302392c185?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-15h,to:now))